src/lightning/data/streaming/item_loader.py

```diff
@@ -146,6 +163,22 @@ def generate_intervals(self) -> List[Tuple[int, int]]:
             begin += num_blocks
         return intervals
 
+    def pre_load_chunk(self, chunk_index: int, chunk_filepath: str) -> None:
```

Comment:
This code is copied from `load_item_from_chunk`. Can you explain why we need to do this? And can we clean it up so there aren't two methods doing almost the same thing?

Reply:
I moved it to a common method. This is called within the prepare chunks thread, so I overlap data loading with data reading and it seems the iteration speed was slightly higher.